SocialEngine/sniffer-rules
# L4 compatibility

### What is the problem / feature ?

- in order to work with laravel 4, we had to use some hacky stuff (aka not really use laravel's artisan command)
- newest version of code sniffer broke the hacks
- while this is Open Source, it had no license or packagist entry
- this was using an incorrect namespace (`Socialengine` instead of `SocialEngine`)

### How did it get fixed / implemented ?

- added a version check and 2 different commands depending on laravel version
- added MIT license
- added keywords and updated description
- renamed a bunch of namespaces and the `ServiceProvider`
- updated the README

### How can someone test / see it ?

- add a [`branch alias`](https://getcomposer.org/doc/articles/aliases.md#require-inline-alias) and repo info to `composer.json`:  

    ```json
    "socialengine/sniffer-rules": "dev-l4-compat as 2.1.0"
    ```
- run `composer update socialengine/sniffer-rules`
- do Laravel 4 steps if you're integrating with L4
- run `php artisan sniff` to ensure everything works
